docs/dapp/tools/verification.md

```diff
@@ -0,0 +1,145 @@
+---
+description: Verifying deployed contracts
+---
+
+# Contract Verification
+
+[Sourcify] is the preferred service for the [verification of smart
+contracts][ethereum-contract-verify] deployed on Sapphire. Make sure you have
+the **address of each deployed contract** available (your deployment scripts
+should report those) and the **contracts JSON metadata file** generated when
+compiling contracts (Hardhat stores it inside the `artifacts/build-info` folder
+and names it as a 32-digit hex number). If your project contains multiple
+contracts, you will need to verify each contract separately.
+
+:::warning Contract deployment encryption
+
+**Do not deploy your contract with an encrypted contract deployment transaction,
+if you want to verify it.** For example, if your `hardhat.config.ts`
+or deployment script contains `import '@oasisprotocol/sapphire-hardhat'` or
+`import '@oasisprotocol/sapphire-paratime'` lines at the beginning, you should
+comment those out for the deployment.
+
+Verification services will try to match the contract deployment transaction code
+with the one in the provided contract's metadata and since the transaction was
```

Comment:
```suggestion
Verification services will try to match the contract deployment transaction code
with the one in the provided contract's metadata. Because the transaction was
```